### ISSUE TITLE
How did you succeed writing registry in the store package?

### ISSUE THREAD

**Issue by Armin2208** [mcdenis/AccentApplicator#4]
Hello,

sorry for misusing the GitHub issues. But it looks like that your program changes the color with changing a value in the registry. You also distribute your app to the Microsoft Store.

I also have a application that edits some registry values. But when I put it in a Windows Application Packaging Project it doesn't can change the registry values. They stay the same.

Could you maybe help me? I really need help but I didn't found a better way to contact you. I really want to put my application also in the Microsoft Store but I can't, due to this issue.

Sorry and thanks and have a nice day!

**Comment by mcdenis** [comment]
Hi,

Accent Applicator only makes changes to the current user session by using an old, but documented system function; It does not edit the registry directly. If you uninstall the app, then reboot your computer, all the changes are gone.

When I developed the app initially, I did want to make the changes persistent by also editing the registry, but I realised that it was not possible as [mentioned](https://docs.microsoft.com/en-us/windows/uwp/porting/desktop-to-uwp-prepare) in the official documentation for the desktop bridge:
> […] entries that your application writes to the HKEY_CURRENT_USER registry hive are placed into a private per-user, per-app location.

In other words, changes made to HKCU are successful from the app's point of view, but are not actually applied to the system. That limitation is on purpose and cannot be overcome as far as I know.

**Comment by Armin2208** [comment]
Hello,

thanks for you answer. I was really happy about it.

I tried to search your code for the part where you change the registry. And I tried to google some alternative ways. I don't find the old system function.

Can you send me the documentation about it/or the code snippet? 

Thank you! <3

**Comment by mcdenis** [comment]
The system function that Accent Applicator uses to apply changes is SetSysColors. Like most functions that apply system changes, it is not available through the Framework Class Library, so it requires P/Invoke. In my app, the function is first defined [here](https://github.com/mcdenis/AccentApplicator/blob/master/AccentCS.Helpers/NativeMethods.vb#L6), then called [here](https://github.com/mcdenis/AccentApplicator/blob/master/AccentCS.Helpers/SystemColors.vb#L178). The official documentation for SetSysColors can be found [here](https://docs.microsoft.com/en-us/windows/desktop/api/winuser/nf-winuser-setsyscolors). Note that although it is officially still supported, in practice, it is not very useful anymore since it can only change system colors and these have been superseded by Visual Style properties (which are not easily customizable). It is why Accent Applicator affects so few UI elements.

Also, as I mentioned, it only applies changes in memory, so the registry is not affected. Some other functions (SystemParametersInfo comes to my mind) do save their changes in the registry, but they can only be used to change a specific subset of system settings.

Ultimately, there is no universal technique (as far as I know) that can be used by Store app to write to the registry. The solution, if there is one, really depends of what you are trying to achieve. Out of curiosity, I took a look at your repositories and saw that you were working on an app that automatically switches between the light/dark theme. Is it what your questions are about?

**Comment by Armin2208** [comment]
Thanks for you detailed answer and taking time for my issue!

Yes, I really want to publish my program to the Store that automatically switches between the dark and light theme :) But as it seems this will be hour-long work without guarantee of success.

**Comment by mcdenis** [comment]
Yeah, sorry, I am afraid I won't  be of any help with this issue. To be honest, I don't think there is a way of changing the "app mode" system wide through a Store app. Editing the registry is likely a non-starter for the reason we discussed and last time I checked, the UWP public API does not offer much in term of system-wide customization (most values are read-only).

Also, since the light/dark switch is a feature new to Windows 10, there is definitely not a 25 years old documented function laying around for it like there is for System Colors with SetSysColors. 😞 You might be able to reverse-engineer the Settings app and find an undocumented method to change the app-mode, but then I am not sure your app would pass Store certification… 

With all that being said, I am not very much into Windows programming these days, so I might very well be wrong. I can only wish you good luck!

**Comment by Armin2208** [comment]
Yeah I got the problem. It is a bit sad, but still thank you for your help. Maybe I will find a way some day. 

Bye!
